<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>745</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>StegoTool - 图片隐写工具</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="topButtonLayout">
      <item>
       <widget class="QPushButton" name="btnOpenImage">
        <property name="text">
         <string>打开图片</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnGenImage">
        <property name="text">
         <string>生成图片</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="cmbGenImageSize"/>
      </item>
      <item>
       <widget class="QPushButton" name="btnEmbed">
        <property name="text">
         <string>嵌入文本</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnExtract">
        <property name="text">
         <string>提取文本</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnEmbedBinary">
        <property name="text">
         <string>嵌入文件</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnExtractBinary">
        <property name="text">
         <string>提取文件</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="bitCountCombo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnSaveImage">
        <property name="text">
         <string>保存图片</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QHBoxLayout" name="mainContentLayout">
      <item>
       <layout class="QVBoxLayout" name="imagePreviewLayout">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>原始图片</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="originalPreview">
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>250</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>400</width>
            <height>300</height>
           </size>
          </property>
          <property name="text">
           <string>拖拽图片到此处或点击&quot;打开图片&quot;</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>处理后图片</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="processedPreview">
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>250</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>400</width>
            <height>300</height>
           </size>
          </property>
          <property name="text">
           <string>嵌入文本后预览</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="textLayout">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>文本内容</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit">
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>500</height>
           </size>
          </property>
          <property name="placeholderText">
           <string>输入要嵌入的文本，或提取后显示在此处...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="2" column="0">
     <layout class="QHBoxLayout" name="bottomInfoLayout">
      <item>
       <widget class="QLabel" name="embedInfoLabel">
        <property name="styleSheet">
         <string notr="true">color: #666666;</string>
        </property>
        <property name="text">
         <string>当前文本大小：0 字节 / 最大支持：0 字节</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="statusLabel">
        <property name="text">
         <string>未加载图片</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="3" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="messageLabel">
        <property name="styleSheet">
         <string notr="true">padding: 5px; font-weight: bold;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnCancelTask">
        <property name="text">
         <string>取消任务</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="actionOpenImage"/>
    <addaction name="actionSaveImage"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>帮助(&amp;H)</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="statusTip">
    <string/>
   </property>
  </widget>
  <action name="actionOpenImage">
   <property name="text">
    <string>打开图片(&amp;O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveImage">
   <property name="text">
    <string>保存图片(&amp;S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>退出(&amp;X)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>关于(&amp;A)</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpenImage</sender>
   <signal>triggered()</signal>
   <receiver>btnOpenImage</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveImage</sender>
   <signal>triggered()</signal>
   <receiver>btnSaveImage</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>790</x>
     <y>10</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_actionAbout_triggered()</slot>
 </slots>
</ui>
